### PR TITLE
3 new functions added to `y_weapondata`

### PR DESCRIPTION
**Pull request by DEntis-T**

### CODE REVIEW

**Comment by DEntis-T** [comment]
OK, it's probably ready to be merged now, unless someone suggests something new on Discord

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 1]
```diff
@@ -22,7 +22,6 @@ Legal:
 	The Initial Developer of the Original Code is Alex "Y_Less" Cole.
 	Portions created by the Initial Developer are Copyright C 2011
 	the Initial Developer. All Rights Reserved.
-
 Contributors:
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc:25]: Why has this part changed?

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 2]
```diff
@@ -87,7 +82,6 @@ Optional plugins:
 	88   `8b d8'   88 ,adPPPPP88 88 88       88      d8""""""""8b   88          88  
 	88    `888'    88 88,    ,88 88 88       88     d8'        `8b  88          88  
 	88     `8'     88 `"8bbdP"Y8 88 88       88    d8'          `8b 88          88  
-
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc:90]: All these mising blank lines in the header.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 3]
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
 	}
 	return WEAPONDATA_INVALID_RETURN;
 }
+
+#define WEAPON_TYPE_UNKNOWN				WeaponType:0
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: Usea an enum.

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 4]
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
...
+#define WEAPON_TYPE_MEELE 				WeaponType:1
+#define WEAPON_TYPE_PISTOL 				WeaponType:2
+#define WEAPON_TYPE_ASSAULTRIFLE 		WeaponType:3
+#define WEAPON_TYPE_SHOTGUN 			WeaponType:4
+#define WEAPON_TYPE_SUBMACHINEGUN 		WeaponType:5
+#define WEAPON_TYPE_RIFLE 				WeaponType:6
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: Should the type numbers not match the slot numbers?  Or if not ohw are they different (please document the difference)?

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 5]
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
...
+#define WEAPON_TYPE_PROJECTILE 			WeaponType:7
+#define WEAPON_TYPE_THROWABLE 			WeaponType:8
+#define WEAPON_TYPE_USEABLE				WeaponType:9
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: `USEABLE` is a terrible name, I know what you mean but all weapons are "useable".  Maybe `EQUIPMENT`?

File: YSI_Game/y_weapondata/y_weapondata_entry.inc [hunk 6]
```diff
@@ -566,3 +560,141 @@ stock Weapon_GetIDFromModelID(modelid)
...
 }
+
+#define WEAPON_TYPE_UNKNOWN				WeaponType:0
+#define WEAPON_TYPE_MEELE 				WeaponType:1
+#define WEAPON_TYPE_PISTOL 				WeaponType:2
+#define WEAPON_TYPE_ASSAULTRIFLE 		WeaponType:3
+#define WEAPON_TYPE_SHOTGUN 			WeaponType:4
+#define WEAPON_TYPE_SUBMACHINEGUN 		WeaponType:5
+#define WEAPON_TYPE_RIFLE 				WeaponType:6
+#define WEAPON_TYPE_PROJECTILE 			WeaponType:7
```
Comment (Y-Less) [inline YSI_Game/y_weapondata/y_weapondata_entry.inc]: `ARTILLERY`

**Review by Y-Less — CHANGES_REQUESTED** [review]

**Comment by DEntis-T** [comment]
@Y-Less Solved the things you've posted, but something doesn't seem to work at a glance.
This code was working fine:
```pawn
new weapid = 33;
    new WeaponType:type = Weapon_GetType(weapid);
    printf("Weapon type of weapon %i is %i", weapid, _:type);
    if(type == WEAPON_TYPE_LONG_RIFLE)
    {
        printf("weapon type is WEAPON_TYPE_LONG_RIFLE");
    }
```
But suddenly it started to give a runtime error with an ID of 6 saying "Invalid instruction". I don't see anything invalid here

**Comment by DEntis-T** [comment]
> @Y-Less Solved the things you've posted, but something doesn't seem to work at a glance. This code was working fine:
> 
> ```pawn
> new weapid = 33;
>     new WeaponType:type = Weapon_GetType(weapid);
>     printf("Weapon type of weapon %i is %i", weapid, _:type);
>     if(type == WEAPON_TYPE_LONG_RIFLE)
>     {
>         printf("weapon type is WEAPON_TYPE_LONG_RIFLE");
>     }
> ```
> 
> But suddenly it started to give a runtime error with an ID of 6 saying "Invalid instruction". I don't see anything invalid here

Maybe the problem is in this part:
```pawn
    if(type == WEAPON_TYPE_LONG_RIFLE)
    {
        printf("weapon type is WEAPON_TYPE_LONG_RIFLE");
    }
```
I'm currently really tired so I will leave this as it is now

**Comment by DEntis-T** [comment]
Nevermind, I'll get it fixed

**Comment by DEntis-T** [comment]
@Y-Less It works now.

**Pull request closed without merging**